main.tf

```diff
@@ -181,22 +289,30 @@ resource "ibm_is_instance" "vsi" {
 resource "ibm_is_floating_ip" "vsi_fip" {
   for_each       = var.enable_floating_ip ? ibm_is_instance.vsi : {}
   name           = "${each.value.name}-fip"
-  target         = each.value.primary_network_interface[0].id
+  target         = var.use_legacy_network_interface ? each.value.primary_network_interface[0].id : each.value.primary_network_attachment[0].virtual_network_interface[0].id
   tags           = var.tags
   access_tags    = var.access_tags
   resource_group = var.resource_group_id
 }
 
 resource "ibm_is_floating_ip" "secondary_fip" {
-  for_each = length(var.secondary_floating_ips) == 0 ? {} : {
-    for interface in local.secondary_fip_list :
+  for_each = var.use_legacy_network_interface ? length(var.secondary_floating_ips) == 0 ? {} : {
+    for interface in local.legacy_secondary_fip_list :
     (interface.name) => interface
-  }
+  } : {}
   name           = each.key
   target         = each.value.target
   tags           = var.tags
   access_tags    = var.access_tags
   resource_group = var.resource_group_id
 }
 
+resource "ibm_is_floating_ip" "latest_secondary_fip" {
```

Comment:
I wasn't sure what "latest_" means here, should this be called "vni_secondary_fip" to make it clear?